### PR TITLE
Improvements on NuttX app. Add tests for multiple use cases. (RDT-1038)

### PR DESCRIPTION
**Pull request by fdcavalcanti**
## Description
This PR moves the `_get_binary_target_info` method to the `NuttxSerial(EspSerial)` class, since it is only used on Espressif Devices.
Also:
- Remove the 'target' option from the app class (it is not required since we can use the target fixture).
- Add some test cases.

## Testing
Added tests for general use and also QEMU.

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [ ] All CI checks (GH Actions) pass.
- [x] Documentation is updated as needed.
- [x] Tests are updated or added as necessary.
- [x] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

**Comment by hfudev** [comment]
the test cases are failing and I'm afraid it's not a false alarm this time. could you help take a look?

**Comment by fdcavalcanti** [comment]
Does the CI support testing on ESP32S3?

File: pytest-embedded-nuttx/tests/test_nuttx.py [hunk 1]
```diff
@@ -42,3 +42,83 @@ def test_nuttx_app_mcuboot(app):
...
+    result = testdir.runpytest(
+        '-s',
+        '--embedded-services', 'nuttx,serial',
+        '--port', '/dev/ttyUSB0'
+    )
+
+    result.assert_outcomes(passed=2)
+
+
+def test_nuttx_app_qemu(testdir):
```
Comment (hfudev) [inline pytest-embedded-nuttx/tests/test_nuttx.py:97]:
```suggestion
qemu_bin_required = pytest.mark.skipif(
    shutil.which('qemu-system-xtensa') is None,
    reason='Please make sure that `qemu-system-xtensa` is in your PATH env var. Build QEMU for ESP32 locally and then '
           'run `pytest` again'
)

@qemu_bin_required
def test_nuttx_app_qemu(testdir):
```

**Comment by hfudev** [comment]
> Does the CI support testing on ESP32S3?

no for ARM runners. the ARM runners only have esp32, esp32s2, and esp32c3 connected.

could you make the tests run with one of these targets? s2 shall also work?

**Comment by fdcavalcanti** [comment]
Ok, I did not know that.
I'll just switch to ESP32, no problem.

**Comment by hfudev** [comment]
Thanks for the PR. Merging it.

**Pull request merged**